### ISSUE TITLE
Eliminating high memory usage

### ISSUE THREAD

**Issue by smogm** [bitbaymarket/bitbay-core#41]
I noticed that the bitbayd is consuming high amounts of memory during its runtime. Compared to other coins with bitcoin core based daemons and similar or larger blockchain size, the memory consumption of bitbayd is unusually high.

As I'm running the bitbayd on my odroid c2 with 2 GB of memory, it would be nice to find out why the usage is that high.
I also played around with the dbcache and mempool sizes, but this didn't change anything.
I noticed that the daemon is using only one single blockfile `blk0001.dat` with currently around  1.2G of space. Maybe it is related to this? Is it possible to reduce the blockfile split size? 

Reserved memory is around 1 GB at runtime,

**Comment by yshurik** [comment]
Yes I have seen such memory usage (a bit more 1GB) with bitbayd. My guesses were just about leveldb actively using ram for caching. Though I had no attempts to check yet. Is that comparable with other coins which have root from peercoin/blackcoin?

**Comment by dzimbeck** [comment]
I think it's common
